DependencyInjection/CompilerPass/PluginPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Http\HttplugBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Remove plugins if their dependencies are missing.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class PluginPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!($container->hasDefinition('logger') || $container->hasAlias('logger'))) {
+            $container->removeDefinition('httplug.plugin.logger');
+        }
+
+        if (!$container->hasDefinition('debug.stopwatch')) {
+            $container->removeDefinition('httplug.plugin.stopwatch');
+        }
+    }
```

Comment:
hm, i wonder if we could instead detect the problem in the compiler pass and throw the exception "to use the logger plugin, you need to have a logger service configured"